client/client.go

```diff
@@ -32,6 +36,12 @@ var (
 
 	whoisClient              = whois.NewClient().WithReferralCache(true)
 	whoisExpirationDateCache = gocache.NewCache().WithMaxSize(10000).WithDefaultTTL(24 * time.Hour)
+
+	mqttTemplateEngine = template.New("base").Funcs(template.FuncMap{
```

Comment:
The only think I'm unsure about is the fact that this uses a templating engine, while everything else leverages placeholder and functions. Maybe this should be exposed as a placeholder instead? I feel supporting `[UUID4]` globally (and not just for MQTT) would be more beneficial, albeit more complicated.

The UUID4 placeholder should also be unique per endpoint, meaning that if it's in the url and the body, it should be the same. Let me know if that doesn't make sense, I can give you some examples

Reply:
Makes sense. I will look around and see if I can just figure it out.

Reply:
Haven’t had time to look into this yet